src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -543,6 +545,24 @@ void IrcMessageHandler::handleUserStateMessage(Communi::IrcMessage *message)
     }
 }
 
+// This will emit only once and right after user logs in to IRC - reset emote data and reload emotes
+void IrcMessageHandler::handleGlobalUserStateMessage(
+    Communi::IrcMessage *message)
+{
+    auto currentUser = getApp()->accounts->twitch.getCurrent();
+
+    // set received emote-sets, this time used to initially load emotes
+    // NOTE: this should always return true unless we reconnect
+    auto emoteSetsChanged = currentUser->setUserstateEmoteSets(
+        message->tag("emote-sets").toString().split(","));
+
+    // We should always attempt to reload emotes even on reconnections where
+    // emoteSetsChanged, since we want to trigger emote reloads when
+    // "currentUserChanged" signal is emitted
+    qCDebug(chatterinoTwitch) << emoteSetsChanged << message->toData();
+    currentUser->loadEmotes();
```

Comment:
If emote sets didn't change we don't want to reload?

Reply:
`emoteSetsChanged` might be false in some very rare cases on very rapid reconnections, however given previous logic, where we triggered `currentUser->loadEmotes();` on `currentUserChanged` signal it'd make sense to attempt reloading emotes here too just in case.
Just to clarify - with old logic, even on those very rapid reconnections we were attempting to reload emotes anyway, as the user has changed from none to whoever you're logged in as.